src/message.ts

```diff
@@ -613,6 +625,45 @@ export class Message implements Sayable {
   // }
 
   /**
+   * @param sendTo UserId/RoomId/Room/Contact
+   */
+  public forward(room: Room): Promise<boolean>
+  public forward(contact: Contact): Promise<boolean>
+  public forward(id: string): Promise<boolean>
+  public forward(sendTo: string|Room|Contact): Promise<boolean> {
+    const m = <MsgRawObj>this.rawObj
+    const newMsg = <MsgRawObj>{}
+    let id = ''
+    if (sendTo instanceof Room || sendTo instanceof Contact) {
+      id = sendTo.id
+    } else if (typeof sendTo === 'string') {
+      id = sendTo
+    }
+    newMsg.ToUserName = id
+    newMsg.FromUserName = config.puppetInstance().userId || ''
+    newMsg.isTranspond = true
+    newMsg.MsgIdBeforeTranspond = m.MsgIdBeforeTranspond || m.MsgId
+    newMsg.MMSourceMsgId = m.MsgId
+
+    // The following parameters need to be overridden after calling createMessage()
+
+    // If you want to forward the file, would like to skip the duplicate upload, sendByLocal must be false.
+    // But need to pay attention to file.size> 25Mb, due to the server policy restrictions, need to re-upload
```

Comment:
Could we reuse the `MediaId` instead of re-upload it?

Because some projects use the `MediaId` directly, which means it might be valid across sessions and accounts.

See:  https://github.com/Chatie/wechaty/issues/422#issuecomment-295564721 

> I think the solution should be not to upload the same media file twice by caching the MediaId of the Uploaded File and reuse it in the future.

> There also has a very interesting project that we could have a look: [微信表情轰炸器](https://github.com/spacelan/wechat-emoticon)

> It can send emoticon by MediaId directly, no need to upload anything.